packages/dds/tree/src/test/feature-libraries/modular-schema/modularChangeFamily.spec.ts

```diff
@@ -1436,3 +1517,55 @@ describe("ModularChangeFamily", () => {
 function treeChunkFromCursor(cursor: ITreeCursorSynchronous): TreeChunk {
 	return chunkTree(cursor, defaultChunkPolicy);
 }
+
+function normalizeChangeset(change: ModularChangeset): ModularChangeset {
+	// const idMappings: ChangeAtomIdMap<NodeId> = new Map();
+	const idAllocator = idAllocatorFromMaxId();
+
+	const nodeChanges: ChangeAtomIdMap<NodeChangeset> = new Map();
+
+	const normalizeNodeChanges = (nodeId: NodeId): NodeId | undefined => {
+		// assert(
+		// 	tryGetFromNestedMap(idMappings, nodeId.revision, nodeId.localId) === undefined,
+		// 	"Found duplicate node ID",
+		// );
+
+		// setInNestedMap(idMappings, nodeId.revision, nodeId.localId, newId);
+
+		const nodeChangeset = tryGetFromNestedMap(
+			change.nodeChanges,
+			nodeId.revision,
+			nodeId.localId,
+		);
+		assert(nodeChangeset !== undefined, "Unknown node ID");
+
+		const normalizedNodeChangeset: NodeChangeset = { ...nodeChangeset };
+		if (normalizedNodeChangeset.fieldChanges !== undefined) {
+			normalizedNodeChangeset.fieldChanges = normalizeFieldChanges(
+				normalizedNodeChangeset.fieldChanges,
+			);
+		}
+
+		const newId: NodeId = { localId: brand(idAllocator.allocate()) };
+		setInNestedMap(nodeChanges, newId.revision, newId.localId, normalizedNodeChangeset);
+
+		return newId;
+	};
+
+	function normalizeFieldChanges(fields: FieldChangeMap): FieldChangeMap {
+		const normalizedFieldChanges: FieldChangeMap = new Map();
+
+		for (const [field, fieldChange] of fields) {
+			const changeHandler = getFieldKind(fieldKinds, fieldChange.fieldKind).changeHandler;
+			normalizedFieldChanges.set(
+				field,
+				changeHandler.rebaser.prune(fieldChange.change, normalizeNodeChanges),
+			);
+		}
+
+		return normalizedFieldChanges;
+	}
+
+	const fieldChanges = normalizeFieldChanges(change.fieldChanges);
+	return { ...change, nodeChanges, fieldChanges };
```

Comment:
This function has the potential to drop entries in `change.nodeChanges` if those do not also appear in the field changes. It's fair to say there should be no such entries, but seeing as this is test code, it seems best not to assume correctness here. You could try to normalize those IDs as well, or (better) you could assert that `change.nodeChanges.size === nodeChanges.size`.